src/features/pdf/data/generatePdfSagas.ts

```diff
@@ -0,0 +1,209 @@
+    )
+    .filter(([_, component]) => !excludedComponents.has(component.id))
+    .map(([pageRef, component]) => {
+      const layoutComponent = getLayoutComponentObject(component.type as ComponentExceptGroupAndSummary);
+
+      if (component.type === 'Group' || layoutComponent?.getComponentType() === ComponentType.Form) {
+        return {
+          id: `__pdf__${component.id}`,
+          type: 'Summary',
+          componentRef: component.id,
+          pageRef,
+          excludedChildren: pdfFormat?.excludedComponents,
+        } as ILayoutCompSummary;
+      }
+      if (layoutComponent?.getComponentType() === ComponentType.Presentation) {
+        return {
+          ...component,
+          id: `__pdf__${component.id}`,
+        };
+      }
+      return null;
+    })
+    .forEach((summaryComponent) => {
+      if (summaryComponent !== null) {
+        automaticPdfLayout.push(summaryComponent);
+      }
+    });
+
+  return automaticPdfLayout;
+}
+
+function* generatePdfSaga(): SagaIterator {
+  try {
+    const layouts: ILayouts = yield select(layoutsSelector);
+    const uiConfig: IUiConfig = yield select(uiConfigSelector);
+    const pdfFormat: IPdfFormat = yield select(pdfFormatSelector);
+    const method: IPdfMethod = yield select(pdfMethodSelector);
+
+    if (method == 'auto') {
+      // Automatic layout
+      const pdfLayout = generateAutomaticLayout(pdfFormat, uiConfig, layouts);
+      yield put(FormLayoutActions.updateLayout({ [PDF_LAYOUT_NAME]: pdfLayout }));
+    }
+
+    yield put(PdfActions.generateFulfilled());
+  } catch (error) {
+    yield put(PdfActions.generateRejected({ error }));
+  }
+}
+
+function* fetchPdfFormatSaga(): SagaIterator {
+  const layoutSets: ILayoutSets = yield select(layoutSetsSelector);
+  const uiConfig: IUiConfig = yield select(uiConfigSelector);
+  const instance: IInstance = yield select(instanceSelector);
+  const applicationMetadata: IApplicationMetadata = yield select(applicationMetadataSelector);
+
+  const dataGuid = getCurrentTaskDataElementId(applicationMetadata, instance, layoutSets);
+  let pdfFormat: IPdfFormat;
+  if (typeof dataGuid === 'string') {
+    try {
+      pdfFormat = yield call(httpGet, getPdfFormatUrl(instance.id, dataGuid));
+    } catch {
+      pdfFormat = {
+        excludedPages: uiConfig.excludePageFromPdf ?? [],
+        excludedComponents: uiConfig.excludeComponentFromPdf ?? [],
+      };
+    }
+  } else {
+    pdfFormat = {
+      excludedPages: uiConfig.excludePageFromPdf ?? [],
+      excludedComponents: uiConfig.excludeComponentFromPdf ?? [],
+    };
+  }
+  yield put(PdfActions.pdfFormatFulfilled({ pdfFormat }));
+}
+
+/**
+ * Watches for changes in formdata and calls fetchPdfFormat and regenerates pdf layout if the method is set to automatic
+ */
+export function* watchPdfPreviewSaga(): SagaIterator {
+  while (true) {
+    yield race([take(FormDataActions.submitFulfilled), take(PdfActions.pdfStateChanged)]);
```

Comment:
The PDF format does rely on the form data, since it is essentially backend dynamics. So if you want to preview automatic mode, and have rules in the pdf-formatter, it would need to refetch this every time the data changes. This is only for preview mode though, so does not affect regular users. It behaved the same way previously, just it was less explicit.